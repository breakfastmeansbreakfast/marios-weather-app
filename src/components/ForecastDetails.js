import React from "react";
import ForecastSummary from "./ForecastSummary";

function ForecastDetails({ forecast }) {
  const { temperature, humidity, wind } = forecast;
  return (
    <div className="forecast-details">
      <div className="forecast-details__date">
        {ForecastSummary.formattedDate}
      </div>
      <div className="forecast-details__description">
        {ForecastSummary.description}
      </div>
      <div className="forecast-details__icon">{ForecastSummary.icon}</div>
      <div className="forecast-details__temperature-max">
        {temperature.max}&deg;C
      </div>
      <div className="forecast-details__temperature-min">
        {temperature.min}&deg;C
      </div>
      <div className="forecast-details__wind-speed">{wind.speed}</div>
      <div className="forecast-details__wind-direction">{wind.direction}</div>
      <div className="forecast-details__humidity">{humidity}</div>
    </div>
  );
}

export default ForecastDetails;
