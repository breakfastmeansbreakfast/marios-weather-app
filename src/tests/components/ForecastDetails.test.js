import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    description: "Stub description",
    icon: "stubIcon",
    temperature: {
      min: 11,
      max: 22,
    },
    wind: {
      speed: 15,
      direction: "n",
    },
    humidity: 29,
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
