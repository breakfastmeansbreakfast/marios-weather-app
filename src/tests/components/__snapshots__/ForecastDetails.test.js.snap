// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    />
    <div
      class="forecast-details__description"
    />
    <div
      class="forecast-details__icon"
    />
    <div
      class="forecast-details__temperature-max"
    >
      22°C
    </div>
    <div
      class="forecast-details__temperature-min"
    >
      11°C
    </div>
    <div
      class="forecast-details__wind-speed"
    >
      15
    </div>
    <div
      class="forecast-details__wind-direction"
    >
      n
    </div>
    <div
      class="forecast-details__humidity"
    >
      29
    </div>
  </div>
</DocumentFragment>
`;
